### PR TITLE
Bash wrapper + systemd service

### PR DESCRIPTION
**Pull request by Nefelim4ag**
Rewrite after your comments:
1. Pure daemon wrapper, no cmd wrapper
2. Config per UUID
3. Almost everything configurable
4. Code check DB size, and alignment for 16MB
5. Code simpler
6. Code don't use lock files
7. Can be easy reused for create cmd simple wrapper

Thanks.

### CODE REVIEW

File: scripts/bees [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+INFO "Mounts base dir: $MOUNT_DIR"
+BASE="/dev/disk/by-uuid/"
+for UUID in "${BTRFS_UUIDs[@]}"; do
+    [ ! -b "$BASE/$UUID" ] && ERRO "Missing: $BASE/$UUID"
+    mkdir -p "$MOUNT_DIR/$UUID"
+    mount -osubvolid=5 "$BASE/$UUID" "$MOUNT_DIR/$UUID" || exit 1
+done
+
+cd "$MOUNT_DIR"
+bees_w ./*-*-*-*-*
```
Comment (Nefelim4ag) [inline scripts/bees]: Then, i prefer to have a subvol inside a root of FS with custom name.
I will implement the code with several configs, i.e. config per FS UUID
Thanks.

File: scripts/bees [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    TMP="$(mktemp -u)"
+    CURRENT_SIZE="$(du -h --apparent-size $DB_PATH | sed 's/\t/ /g' | cut -d' ' -f1)"
+    NEW_SIZE="$DB_SIZE"
+    {
+        echo $NEW_SIZE
+        echo $CURRENT_SIZE
+    } | sort -h | tail -n 1 > "$TMP"
+    read -r BIGGER_SIZE < "$TMP"
+
+    [ "$BIGGER_SIZE" != "$NEW_SIZE" ] && ERRO "Can't shrink db file"
```
Comment (Nefelim4ag) [inline scripts/bees]: Thanks, I will change code for check size only for 16MB alignment

**Comment by Nefelim4ag** [comment]
Does it possible to disable internal log timestamps?
/* As i use and test bees by systemd unit, i already have timestamps in logs */

**Comment by Zygo** [comment]
I can certainly add that feature.  I plan to add a proper argument parser:

 - -b BEESHOME - set BEESHOME path for following directory arguments
 - -c SIZE - create hash table of given size or resize hash table
 - -l FILE - write log to file instead of stdout/stderr
 - -t full/relative/off - timestamp format (absolute time, seconds-since-start, or none)

-b is relative to the root of the filesystem if it does not start with '/'.  -s is relative to -b.  So you could write

        bees -b beeshome -c 1g -l log.txt -t full /path1 /path2 /path3

to get something like my setup, while a systemd setup would look more like

        bees -b .beeshome -c 1g -t off /path1 /path2 /path3

If you wanted all the hash tables in one place you would use a full path:

        bees -b /bigdisk/hashtables -c 50g -t off /path1 /path2 /path3

This change would come with some tweaks to the file layout so that one daemon can process multiple filesystems:

 - bees would try to open beeshash.UUID.dat first, then fall back to beeshash.dat if it didn't exist
 - similar for log files (or hardcode the name in beeshome, so -l is just an on/off switch)

Shortening the paths in the log messages is harder than it looks because the path is part of the thread name.  The status file dumps the state of all the threads in the process, but maybe I can rework that so that each context has its own status object and reports only its own threads.

**Comment by Nefelim4ag** [comment]
@Zygo,
What reason to support handle several FS in one command, if Bees have a problem with shared dat file?
I understand what in you example you assume that with relative path, bees will autocreate dat file & etc for each FS, it's ok, but i don't see any pros for that. it's only add some complexity to code.
/* I think it's more easy to track several processes compare to threads in one big process */

---
And by your comment:
```
The hash table currently cannot be shared across filesystems. That capability was an experimental feature left over from some earlier prototyping. The experiment didn't end well.

I abandoned that feature because it causes a performance disaster.
```
I just think what you will drop support for that.

**Comment by Zygo** [comment]
Now that we're explicitly not supporting BEESHOME sharing any more, there can be locks on the data files.  A simple `flock(hash_table_fd, LOCK_EX|LOCK_NB)` is probably sufficient for that.

The BEESHOME directory can no longer be shared between BeesContexts.  It is no longer necessary to put the UUID in the beescrawl.dat filename.  bees will now read the old crawl file name if it exists, and rename to the new one.

The BEESHOME environment variable can now be a relative path from the filesystem root, so multiple BeesContext instances can still be created in the same process as long as BEESHOME does not begin with '/'.  They share FD cache and event counters, but in every other respect it is now the same result as if one bees process had been created per filesystem.

I'm not sure if it's worth keeping the ability to dedup more than one filesystem per process--I certainly don't recommend it--but at this moment it's extra work to remove that capability.  It may disappear when I get to the command line option parsing changes coming up.

I plan to handle hash table creation and resizing in bees eventually, but didn't finish that yet (I got as far as adding parameters to the constructor).

I plan to transition to using command line options for configuration, where options take precedence over existing environment variables.  I plan to have no new environment variables for configuration of bees (though there might be some for the internals of libcrucible).

At some point it would be nice to implement a configuration file so all the hardcoded time intervals and rates in bees.h could be changed without rebuilding the binary.

**Pull request merged**